advanced_alchemy/repository/memory/_sync.py

```diff
@@ -210,36 +210,60 @@ def _exclude_unused_kwargs(self, kwargs: dict[str, Any]) -> dict[str, Any]:
     def _apply_limit_offset_pagination(result: list[ModelT], limit: int, offset: int) -> list[ModelT]:
         return result[offset:limit]
 
-    @staticmethod
+    def _extract_field_name(self, field: "Union[str, ColumnElement[Any], InstrumentedAttribute[Any]]") -> str:
+        """Extract string field name from various input types.
+
+        Args:
+            field: Field name, column element, or instrumented attribute
+
+        Returns:
+            str: String field name for use with getattr()
+
+        Raises:
+            RepositoryError: If a ColumnElement (func expression) is used with mock repository
+        """
+        if isinstance(field, str):
+            return field
+        if isinstance(field, InstrumentedAttribute):
+            # Extract the attribute name from InstrumentedAttribute
+            return field.key
+        # ColumnElement (func expressions) cannot be executed in-memory
+        msg = "func expressions and column elements are not supported in mock repositories (in-memory filtering)"
```

Comment:
Are we sure these are `func` expressions at this point? It could also be any other unsupported type iiuc

Reply:
Yes, I think that would be good. Maybe report the actual type that was received and is unsupported. e.g. `{type(field)} columns are not supported...`